crates/ruff/src/rules/tryceratops/rules/raise_within_try.rs

```diff
@@ -77,15 +82,55 @@ pub(crate) fn raise_within_try(checker: &mut Checker, body: &[Stmt], handlers: &
         return;
     }
 
+    // The names of exceptions handled by the `try` Stmt. We can't compare `Expr`'s since they have
+    // different ranges, but by virtue of this function's call path we know that the `raise`
+    // statements will always be within the surrounding `try`.
+    let handler_ids: FxHashSet<&str> = helpers::extract_handled_exceptions(handlers)
+        .into_iter()
+        .filter_map(|handler| {
+            if let Expr::Name(ast::ExprName { id, .. }) = handler {
+                Some(id.as_str())
+            } else {
+                None
+            }
+        })
+        .collect();
+
     let raises = {
         let mut visitor = RaiseStatementVisitor::default();
         visitor.visit_body(body);
         visitor.raises
     };
 
     for stmt in raises {
-        checker
-            .diagnostics
-            .push(Diagnostic::new(RaiseWithinTry, stmt.range()));
+        let Stmt::Raise(ast::StmtRaise { exc, .. }) = stmt else {
+            continue;
+        };
+
+        let Some(exception) = exc else {
+            continue;
+        };
+
+        let exc_name = get_function_name(exception.as_ref()).unwrap_or_default();
```

Comment:
Should we just ignore the `exception` if `get_function_name` returns `None` or is there a case where we want to lookup `""` in `handler_ids`?

Reply:
Yeah, could also just ignore it - there's a case where `raise` with no arguments raises the last handled arg, but I guess the `exc` argument there is `None` rather than `""`. Sorry, should've checked that